### PR TITLE
Remove `nx-cugraph` build/test/publish from cugraph

### PR DESCRIPTION
**Pull request by nv-rliu**
Closes https://github.com/rapidsai/graph_dl/issues/646

As part of the repo migration effort, this PR removes steps relevant to build/test/publish of `nx-cugraph` from this repo.

Changes are ready but just to be safe **DO NOT MERGE** until we begin publishing packages from the new repo.

### CODE REVIEW

File: ci/test_wheel.sh [hunk 1]
```diff
@@ -41,6 +37,6 @@ else
        -v \
        --import-mode=append \
        --benchmark-disable \
-       -k "not test_property_graph_mg and not test_bulk_sampler_io" \
-       ./python/${package_name}/${python_package_name}/tests
+       -k "test_pagerank_non_convergence" \
+       ./python/${package_name}/${python_package_name}/tests/link_analysis/
```
Comment (eriknw) [inline ci/test_wheel.sh:41]: Why was this changed?

Also, we can probably delete `NX_CUGRAPH_USE_COMPAT_GRAPHS=False` a few lines down.

Comment (nv-rliu) [reply ci/test_wheel.sh:41]: Ah, sorry. This was not a change intended for this PR

**Review by jameslamb — APPROVED** [review]
Great! This looks perfect, this should be enough to ensure we don't publish any more `nx-cugraph` 24.12 nightlies to the nightly PyPI index / `rapidsai-nightly` conda channel.

Up to you two, but I'd wait to merge this until we're ready to publish from the `nx-cugraph` repo.

**Review by raydouglass — APPROVED** [review]

**Review by bdice — APPROVED** [review]

**Review by eriknw — APPROVED** [review]
Can we delete this FIXME now and `PY_IGNORE_IMPORTMISMATCH=1`? Or is this needed for other libraries?
https://github.com/rapidsai/cugraph/blob/2d1189b82b134cd5e319b0a6456ef723c2ac8de6/ci/test_wheel.sh#L30-L34

Also, for posterity / reminder-to-self, docs are still in the `cugraph` repo:
https://github.com/rapidsai/cugraph/blob/2d1189b82b134cd5e319b0a6456ef723c2ac8de6/ci/release/update-version.sh#L121

I don't consider these blockers, so I'm going ahead and approving :rocket:
